Mathlib/FieldTheory/Galois/Infinite.lean

```diff
@@ -0,0 +1,296 @@
+    intro y hy
+    simp only [AlgEquiv.smul_def, AlgEquiv.mul_apply]
+    have : ((restrictNormalHom L') h ⟨y,hy⟩).1 = ((restrictNormalHom L') σ ⟨y,hy⟩).1 := by rw [eq]
+    rw [restrictNormalHom_apply L'.1 h ⟨y, hy⟩, restrictNormalHom_apply L'.1 σ ⟨y, hy⟩] at this
+    simp only [this, ← AlgEquiv.mul_apply, inv_mul_cancel, one_apply]
+  absurd compl
+  apply Set.not_subset.mpr
+  use h
+  simpa only [this, Set.mem_compl_iff, Subsemigroup.mem_carrier, Submonoid.mem_toSubsemigroup,
+    Subgroup.mem_toSubmonoid, not_not, true_and] using mem
+
+/-- The Galois correspondence from intermediate fields to ClosedSubgroup. -/
+def IntermediateFieldEquivClosedSubgroup [IsGalois k K] :
+    IntermediateField k K ≃o (ClosedSubgroup (K ≃ₐ[k] K))ᵒᵈ where
+  toFun := fun L =>
+    { L.fixingSubgroup with
+      isClosed' := fixingSubgroup_isClosed L }
+  invFun := fun H => IntermediateField.fixedField H.1
+  left_inv := fun L => fixedField_fixingSubgroup L
+  right_inv := by
+    intro H
+    simp_rw [fixingSubgroup_fixedField H]
+    rfl
+  map_rel_iff' := by
+    intro L₁ L₂
+    show L₁.fixingSubgroup ≥ L₂.fixingSubgroup ↔ L₁ ≤ L₂
+    rw [← fixedField_fixingSubgroup L₂, IntermediateField.le_iff_le, fixedField_fixingSubgroup L₂]
+
+/-- The Galois correspondence as a `GaloisInsertion` -/
+def GaloisInsertionIntermediateFieldSubgroup [IsGalois k K] :
+    GaloisInsertion (OrderDual.toDual ∘ fun (E : IntermediateField k K) ↦
+      (⟨E.fixingSubgroup, fixingSubgroup_isClosed E⟩ : ClosedSubgroup (K ≃ₐ[k] K)))
+      ((fun (H : ClosedSubgroup (K ≃ₐ[k] K)) ↦ IntermediateField.fixedField H) ∘
+        OrderDual.toDual) where
+  choice E _ := ⟨E.fixingSubgroup, fixingSubgroup_isClosed E⟩
+  gc E H := (IntermediateField.le_iff_le H.1 E).symm
+  le_l_u H := by
+    simp only [Function.comp_apply, fixingSubgroup_fixedField]
+    rfl
+  choice_eq _ _ := rfl
+
+/-- The Galois correspondence as a `GaloisCoinsertion` -/
+def GaloisCoinsertionIntermediateFieldSubgroup [IsGalois k K] :
+    GaloisCoinsertion (OrderDual.toDual ∘ fun (E : IntermediateField k K) ↦
+      (⟨E.fixingSubgroup, fixingSubgroup_isClosed E⟩ : ClosedSubgroup (K ≃ₐ[k] K)))
+      ((fun (H : ClosedSubgroup (K ≃ₐ[k] K)) ↦ IntermediateField.fixedField H) ∘
+        OrderDual.toDual) where
+  choice H _ := IntermediateField.fixedField H.1
+  gc E H := (IntermediateField.le_iff_le H.1 E).symm
+  u_l_le K := le_of_eq (fixedField_fixingSubgroup K)
+  choice_eq _ _ := rfl
+
+private lemma IntermediateField.finiteDimensional_of_le {M N : IntermediateField k K}
+    (le : M ≤ N) [FiniteDimensional k N] : FiniteDimensional k M := by
+  let i : M →ₐ[k] N := {
+    toFun := fun x => ⟨x.1, le x.2⟩
+    map_one' := rfl
+    map_mul' := fun _ _ => rfl
+    map_zero' := rfl
+    map_add' := fun _ _ => rfl
+    commutes' := fun _ => rfl }
+  apply FiniteDimensional.of_injective (AlgHom.toLinearMap i)
+  intro _ _ h
+  apply_fun Subtype.val at h
+  exact Subtype.val_injective h
```

Comment:
This is worth moving to an earlier file. You could use `IntermediateField.inclusion` to help golf this.

Reply:
Thank you for your suggestion, but I didn't find a good place for it, `find_home` suggest `FieldTheory.Adjoin`, but I didn't see places relavant with it, any suggestions about where to put it?

Reply:
Yeah, maybe either adjoin or the relrank file (where you could probably also prove this using `IntermediateField.relrank_dvd_of_le_left`).